src/yarpserver/src/main.cpp

```diff
@@ -5,9 +5,15 @@
  *
  */
 
+#include <stdio.h>
+#include <yarp/os/Network.h>
 #include <yarp/serversql/yarpserversql.h>
 
-int main(int argc, char *argv[]) {
+int main(int argc, char *argv[])
+{
+    // Yarp server must always run using system clock
+    yarp::os::Network::setEnvironment(yarp::os::ConstString("YARP_CLOCK"), yarp::os::ConstString(""));
```

Comment:
Yes, kind of. Resetting the env var in this way is a dirty trick.
The idea is to have an instance of clock which is alway system clock and a different clock which is 'mutable' depending on the env var. The system clock will be used in all places which should always rely on a deterministic clock like yarpserver in this case (i.e. makes no sense for it to wait for a network clock).
Right now it is like we have only the mutable one so all thread will be affected by the env var, also yarpserver itself.